libdino/src/entity/account.vala

```diff
@@ -73,6 +69,38 @@ public class Account : Object {
         notify.connect(on_update);
     }
 
+    public async string? get_password() {
+#if WITH_SECRET
+        if (password != "") {
+            var cleartext_password = password;
+            // We have a clear-text password.  Try to put it into the keyring.
+            if (yield secret_manager.set_password(password)) {
+                // Success.  Clear the clear-text PW from this entity.
+                this.password = "";
+            }
+            // Either way, return the password.
+            return cleartext_password;
+        } else {
+            // We have to look it up in the secret service
+            return yield secret_manager.get_password();
+        }
+#else
+        return this.password;
+#endif
+    }
+
+    public async void set_password(string password) {
+#if WITH_SECRET
+        if (yield secret_manager.set_password(password)) {
+            // Success.  Don't keep anything locally.
+            this.password = "";
+        } else
```

Comment:
Here too you can replace the else with a return inside of the if.

Reply:
I usually prefer else of early-return in cases like this, but with the additional preprocessor conditional I agree that's clearer here.